elastic/beats
# [Elastic-Agent] IUse itnernal port for local fleet server

## What does this PR do?

What this PR solves is a problem when agent got unenrolled on heavier load when agent managing fleet server cannot checkin to it's own server so it will fallback to unenroll.

Problem is solved by adding internal endpoint which is used for communication on local network (with agent handling fleet server)
It lets FS to spin up 2 set of handlers, one on public 8220 and one on port defined in config. 

## Why is it important?

Prevents unwanted un-enrolls of the most important agent in deployment

## Checklist



## How to test this PR locally

This needs to be tested with work on `fleet-server` Link: https://github.com/elastic/fleet-server/pull/880
- Start stack
- Install agent with FS in a policy
- Check ports
```
sh-3.2# lsof -i -P | grep LISTEN | grep fleet
fleet-ser  7056            root   19u  IPv4 0xba7881a9227099a5      0t0    TCP localhost:{random_port} (LISTEN)
fleet-ser  7056            root   21u  IPv6 0xba7881a91284721d      0t0    TCP *:8220 (LISTEN)
```

- run wireshark, set filter to random port, there should be some comm
- set filter to `8220` port, there should be no comm
- enroll new agent, from another VM
- there should be some comm on both ports
